### PR TITLE
[CARBONDATA-1669] Clean up code in CarbonDataRDDFactory

### PR DESCRIPTION
**Pull request by jackylk**
Inside CarbonDataRDDFactory.loadCarbonData, there are many function defined inside function, makes the loading logic very hard to read
This PR improves its readability

 - [X] Any interfaces changed?
 No

 - [X] Any backward compatibility impacted?
 No

 - [X] Document update required?
No

 - [X] Testing done
No new testcase is required

 - [X] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
NA

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/811/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1444/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/818/

**Comment by jackylk** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/821/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1450/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1451/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1454/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1472/

**Comment by jackylk** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/840/

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/AlterTableCompactionCommand.scala [hunk 1]
```diff
@@ -87,4 +94,113 @@ case class AlterTableCompactionCommand(
...
+      compactionSize = CarbonDataMergerUtil.getCompactionSize(CompactionType.MAJOR_COMPACTION)
+      compactionType = CompactionType.MAJOR_COMPACTION
+    } else if (alterTableModel.compactionType
+      .equalsIgnoreCase(CompactionType.IUD_UPDDEL_DELTA_COMPACTION.toString)) {
+      compactionType = CompactionType.IUD_UPDDEL_DELTA_COMPACTION
+      if (alterTableModel.segmentUpdateStatusManager.get != None) {
+        carbonLoadModel
+          .setSegmentUpdateStatusManager(alterTableModel.segmentUpdateStatusManager.get)
+
+        carbonLoadModel
```
Comment (QiangCai) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/AlterTableCompactionCommand.scala]: I prefer to :
 carbonLoadModel.setLoadMetadataDetails(
    alterTableModel.segmentUpdateStatusManager.get.getLoadMetadataDetails.toList.asJava)

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1473/

File: integration/spark-common/src/main/scala/org/apache/carbondata/api/CarbonStore.scala [hunk 2]
```diff
@@ -79,16 +82,45 @@ object CarbonStore {
       dbName: String,
       tableName: String,
       storePath: String,
-      carbonTable: CarbonTable, forceTableClean: Boolean): Unit = {
+      carbonTable: CarbonTable, forceTableClean: Boolean
```
[the remark this answers is not in the document]
Comment (jackylk) [reply integration/spark-common/src/main/scala/org/apache/carbondata/api/CarbonStore.scala]: fixed

File: integration/spark-common/src/main/scala/org/apache/carbondata/api/CarbonStore.scala [hunk 3]
```diff
@@ -79,16 +82,45 @@ object CarbonStore {
...
+  ): Unit = {
```
[the remark this answers is not in the document]
Comment (jackylk) [reply integration/spark-common/src/main/scala/org/apache/carbondata/api/CarbonStore.scala]: fixed

File: integration/spark-common/src/main/scala/org/apache/carbondata/api/CarbonStore.scala [hunk 4]
```diff
@@ -79,16 +82,45 @@ object CarbonStore {
...
+      val identifier = new CarbonTableIdentifier(dbName, tableName, "")
+      carbonCleanFilesLock =
+        CarbonLockFactory.getCarbonLockObj(identifier, LockUsage.CLEAN_FILES_LOCK)
     } catch {
       case ex: Exception =>
         sys.error(ex.getMessage)
+      return
     }
-    Seq.empty
+    try {
```
[the remark this answers is not in the document]
Comment (jackylk) [reply integration/spark-common/src/main/scala/org/apache/carbondata/api/CarbonStore.scala]: fixed

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/AlterTableCompactionCommand.scala (hunk 1 above)
Comment (jackylk) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/AlterTableCompactionCommand.scala]: fixed

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/842/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1475/

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/DataManagementFunc.scala [hunk 5]
```diff
@@ -49,105 +44,6 @@ object DataManagementFunc {
 
   private val LOGGER = LogServiceFactory.getLogService(this.getClass.getCanonicalName)
 
-  def deleteLoadByDate(
```
Comment (chenliang613) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/DataManagementFunc.scala]: Just confirm  :   this method is for spark 1.6?

Comment (jackylk) [reply integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/DataManagementFunc.scala]: No one use this function

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/846/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1479/

**Comment by QiangCai** [comment]
LGTM

**Pull request closed without merging**